subprojects/scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java

```diff
@@ -163,7 +163,9 @@ protected ScalaJavaJointCompileSpec createSpec() {
             ? ImmutableList.of()
             : ImmutableList.copyOf(compileOptions.getAnnotationProcessorPath()));
         spec.setBuildStartTimestamp(getServices().get(BuildStartedTime.class).getStartTime());
-        configureExecutable(spec.getCompileOptions().getForkOptions());
+        if (getOptions().getForkOptions().getExecutable() == null) {
+            configureExecutable(spec.getCompileOptions().getForkOptions());
```

Comment:
I think this might not be quite what we want. This means that `forkOptions.executable` is a higher precedence than `javaLauncher` when set. @ljacomet is this what you had in mind here?

It looks like we have the same issue in GroovyCompile. Can you try adding a similar test for Groovy and see if it breaks in the same way?

Reply:
Nevermind, Louis pointed out to me that we don't allow both forkOptions.executable _and_ javaLauncher to be set.

Can you still check if GroovyCompile needs to be fixed in the same way?

For style, I think it would make sense to move this null check into `configureExecutable()`

Reply:
The blind set done in the `else` branch was actually what happened before: you could not set the Java executable ... because "reasons".
See 202e1040c2e09aadbb4409648672d272d9b807cd where I introduced the toolchain support.

Reply:
OK, I talked with Louis about this a bit more.

It looks like we've always set the executable to the current JVM and we didn't provide a way to change it.

Which test was failing for you that required this change?

Reply:
This one: https://ge.gradle.org/s/5vev64mnkfmji/tests/:scala:embeddedIntegTest/org.gradle.scala.compile.ZincScalaCompilerIntegrationTest/respects%20fork%20options%20settings%20and%20executable%20%5B2.12.10%5D?top-execution=1

Reply:
This is what the Groovy plugin does: https://github.com/gradle/gradle/blob/ade90b458c8b883a8beee992119b88c8435441bd/subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java#L309-L315

And it has a test that explicitly asserts this:
https://github.com/gradle/gradle/blob/bdcf8e3861b305f80d6186da9b535c1fbb10c173/subprojects/plugins/src/integTest/groovy/org/gradle/groovy/compile/DaemonGroovyCompilerIntegrationTest.groovy#L48

I adjusted the test in Scala plugin to assert for this and removed the change I did.
I lack the context to tell how it is "supposed to work" - let me know if instead Groovy part should be "fixed".